docs/source/tutorials/configure_data.rst

```diff
@@ -0,0 +1,372 @@
+                            "type": "huggingface_pre_process",
+                            "params": {
+                                "model_name": "Intel/bert-base-uncased-mrpc",
+                                "input_cols": [
+                                    "sentence1",
+                                    "sentence2"
+                                ],
+                                "label_cols": [
+                                    "label"
+                                ],
+                                "max_samples": null
+                            }
+                        },
+                        "post_process_data": {
+                            "name": "_text_classification_post_process",
+                            "type": "text_classification_post_process",
+                            "params": {}
+                        },
+                        "dataloader": {
+                            "name": "_default_dataloader",
+                            "type": "default_dataloader",
+                            "params": {
+                                "batch_size": 1
+                            }
+                        }
+                    },
+                }
+
+        .. tab:: Python Class
+
+            .. code-block:: python
+
+                from olive.data.config import DataConfig
+                data_config = DataConfig(
+                    name="data",
+                    type="DataContainer",
+                    components={
+                        "load_dataset": {
+                            "name": "_huggingface_dataset",
+                            "type": "huggingface_dataset",
+                            "params": {
+                                "data_dir": null,
+                                "data_name": "glue",
+                                "subset": "mrpc",
+                                "split": "validation",
+                                "data_files": null
+                            }
+                        },
+                        "pre_process_data": {
+                            "name": "_huggingface_pre_process",
+                            "type": "huggingface_pre_process",
+                            "params": {
+                                "model_name": "Intel/bert-base-uncased-mrpc",
+                                "input_cols": [
+                                    "sentence1",
+                                    "sentence2"
+                                ],
+                                "label_cols": [
+                                    "label"
+                                ],
+                                "max_samples": null
+                            }
+                        },
+                        "post_process_data": {
+                            "name": "_text_classification_post_process",
+                            "type": "text_classification_post_process",
```

Comment:
we need add the customized dataset registration in doc like https://github.com/microsoft/Olive/blob/main/test/unit_test/test_data_root.py#L31
https://github.com/microsoft/Olive/blob/main/test/unit_test/test_data_root.py#L107

Reply:
https://github.com/microsoft/Olive/blob/a65b453ba7d15d5b811281c6facb11a9fcc65ec3/examples/snpe/inception_snpe_qualcomm_npu/user_script.py#L9

![image](https://github.com/microsoft/Olive/assets/13343117/08f1d113-59f1-4af1-be9a-1be787fa9ceb)

Is this fine?

Reply:
Better add a section so user could easily find how to customize the data config. Otherwise, he might need search at doc to find how.

Reply:
Updated.